Update README

samrjack opened the pull request:
Here's the start of the README update that goes along with the changes for [issue 35](https://github.com/emacs-tree-sitter/ts-fold/issues/35). I added more details about how people could go about adding their own customizations to the package. Please look it over for correctness and style. I tried to add more tutorial-style explanations to make it easier to approach. If you want to see it rendered out, you can see it on the [main page of my fork](https://github.com/samrjack/ts-fold)

jcs090218 commented on README.md:
```diff
@@ -66,139 +69,222 @@ or
-| `ts-fold-toggle`           | toggle the syntax node at `point'.                                          |
+| `ts-fold-close-all` | close all foldable syntax nodes in the current buffer. |
+| `ts-fold-open-all` | open all folded syntax nodes in the current buffer. |
+| `ts-fold-toggle` | toggle the syntax node at `point'. |
```
I think it's better if we have the table aligned.

samrjack replied:
oh, ew, totally agree. I had prettier running on save but it seems to not like tables. Will disable and fix up.

jcs090218 commented on README.md:
````diff
@@ -66,139 +69,222 @@ or
-   (use-package ts-fold-indicators
-   :straight (ts-fold-indicators :type git :host github :repo "emacs-tree-sitter/ts-fold"))
-   ```
+### :white_circle: Folding on new nodes
````
I prefer unicode here -> ⚪.

samrjack replied:
Added here and and moved the one other occurence of github styling to unicode.

jcs090218 commented on README.md:
````diff
@@ -66,139 +69,222 @@ or
-   ```el
-   (add-hook 'tree-sitter-after-on-hook #ts-fold-indicators-mode)
-   ```
+- `ts-fold-range-seq` - Folds from the start of the node to the end of the node leaving a buffer of one character on each side. Usually used for code blocks that have bracketing delimiters.
````
I prefer max 80 columns per line, can we avoid these long lines?

Sorry, I should have added the `.dir-locals.el`. 😅

samrjack replied:
No problem, I usually right in org mode so I just let the lines go as long as they want 😅.

jcs090218 commented on README.md:
````diff
@@ -66,139 +69,222 @@ or
-```
-M-x ts-fold-indicators-mode
-```
+To find what node you'll want to fold closed, refer to the (tree-sitter documentation)[https://emacs-tree-sitter.github.io/getting-started/#view-the-syntax-tree] about viewing nodes. `tree-sitter-debug` and `tree-sitter-query-builder` are both very useful for this.
````
There is syntax error with the link for `tree-sitter documentation`, should be another way around.

```
[tree-sitter documentation](...)
```

samrjack replied:
Ah, great catch!

jcs090218 commented on README.md:
````diff
@@ -66,139 +69,222 @@ or
-           ((memq ln line-reminder--saved-lines) 'line-reminder-saved-sign-face)
-           (t nil)))))
-   ```
+  / <S> * The main function that gets run after program is compiled *
````
I think this should be `/* <S> The main function that gets run after program is compiled */`.

samrjack replied:
My bad, my personal fold got reapplied when I was testing. However, when I just re-tested, I got 
`/* <S> * The main function that gets run after program is compiled *`. I'll update it to that for now. Can you confirm if that second asterisk is there for you too?

jcs090218 replied:
I have tested in a `c++` file, the second asterisk isn't there... 😕 Just simply `/* <S> The main function that gets run after program is compiled */`.

samrjack replied:
Hmm, strange but oh well 😆. Maybe something in my configuration is messing with it. I'll update the asterisk real quick.

jcs090218 replied:
Yeah, this isn't blocking, and feel free to open up issues if that persist. It would be great if we can fix that! 😅

samrjack replied:
Eh, I don't think it's worth diving into for the time being since there are other places for effort to be spent. And it really isn't blocking or troublesome 😛

jcs090218 commented on README.md:
```diff
@@ -66,139 +69,222 @@ or
+### 📇 Commands
 
-| Commands                   | Description                                                                 |
+The following are the functions provided by `ts-fold-mode`
```
Maybe add a newline after this line so it's not right next to the table.

samrjack replied:
Oh, this was the reason for the table not formatting properly! Thank you for pointing this out!

jcs090218 reviewed (commented):
Some nits.

Thank you so much! ❤️ 🎉 I think the overall quality is way better than the old one. I've always had a hard time documenting stuff, so this is great! 😅 👍

jcs090218 commented on README.md:
```diff
@@ -66,139 +69,264 @@ or
+  }
+
+  // |
+  // | '(comment . ts-fold-range-block-comment)
```
In `c++`, this should be `ts-fold-range-c-like-comment`. Are you using this for the example? 😕 

https://github.com/emacs-tree-sitter/ts-fold/blob/0bae1476a47b707c6003becd87f4c0bf6cf509ba/ts-fold-parsers.el#L84

Is that the reason why we get the second asterisk? 😆

jcs090218 replied:
Oh, I guess it was intentional. Sorry for the noise.

jcs090218 commented:
I think you have removed [📝 Summary](https://github.com/emacs-tree-sitter/ts-fold#-summary) and [⚖️ Indicators Mode](https://github.com/emacs-tree-sitter/ts-fold#%EF%B8%8F-indicators-mode) sections, do you think it's useful? not useful? Would it be better if we keep it? etc. LMKWYT?

samrjack commented:
Oh shoot! I'm glad you noticed. I moved those sections to a single plugins section but you're right that it isn't there... 😅 I'm guessing at some point when moving around it got highlighted and deleted... I'll find it real quick and put it back in.

samrjack commented:
Oh my goodness, I'm so glad emacs had such a long undo history 😂 I was writing a part of that section and then accidentally deleted it when I moved back up to check some other stuff. Didn't even notice it since I was trying to get it wrapped up. I am so glad you noticed!

jcs090218 commented on README.md:
```diff
@@ -14,26 +15,35 @@ to provide code folding based on the tree-sitter syntax tree.
 
 <!-- Markdown is not able to render links with unicode so after refreshing the toc, select it and:
-    `M-x regexp-replace #[^a-zA-Z] <ret> # <ret>` -->
+    `M-x regexp-replace #[^-a-zA-Z] <ret> # <ret>` -->
```
Can you confirm the change here? It seems like you have accidentally changed the line.

samrjack replied:
I added this extra dash because then if you run the search-replace multiple times, it won't clear out the dashes that come after the '#'. By putting it in the front of the matching group, it isn't seen as a range so it is read as a dash.

One thing I just noticed, in your emacs is the command actually `regexp-replace`? In mine, it's showing up as `replace-regexp`. I use evil mode so I I'm not super familiar with standard emacs commands and don't want to change it without checking first.

jcs090218 replied:
>I added this extra dash because then if you run the search-replace multiple times, it won't clear out the dashes that come after the '#'. By putting it in the front of the matching group, it isn't seen as a range so it is read as a dash.

Thanks for the clarification! 👍 

That line was added by @mattiasdrp, but I think you are correct. It shows up `replace-regexp` and not `regexp-replace`. Can you fix that up for us? 😓 Otherwise, it should be good to merge! 😅 Thank you for your patient, sorry it took this long to get this merge!

jcs090218 commented:
One last nit! No worries, that's why I am here to review and take responsibility for this PR. :)

Thank you for taking these sections back! 😁 I thought these parts weren't too helpful, so you removed them!

I am glad you have picked the emoji for each section; I appreciate it! ❤️ 👍

samrjack commented:
I'm probably going to be logging off soon. I updated the regexp command after looking it up in the emacs manual. If you have any more comments before merging, just leave them here and I'll take a look in the morning!

jcs090218 commented:
> I'm probably going to be logging off soon. I updated the regexp command after looking it up in the emacs manual. If you have any more comments before merging, just leave them here and I'll take a look in the morning!

It's good to go! Thank you for being patient with us! 😅 And sorry it took so long to get this merge! It's been great to work with you! 😄 👍

samrjack commented:
Looking forward to providing more contributions in the future ☺️.
